ecs-agent/acs/session/attach_resource_responder.go

```diff
@@ -78,8 +80,17 @@ func (r *attachResourceResponder) handleAttachMessage(message *ecsacs.ConfirmAtt
 	}
 
 	messageID := aws.StringValue(message.MessageId)
-	expiresAt := receivedAt.Add(
-		time.Duration(aws.Int64Value(message.WaitTimeoutMs)) * time.Millisecond)
+	// Set a default wait timeout (5m) for the attachment message from ACS if not provided.
+	// For example, the attachment payload for the EBS attach might not have the property.
+	waitTimeoutMs := aws.Int64Value(message.WaitTimeoutMs)
+	if waitTimeoutMs == 0 {
+		waitTimeoutMs = DefaultAttachmentWaitTimeoutInMs
+		logger.Debug("No waitTimeoutMs is received, set to the default timeout",
+			logger.Fields{
+				"WaitTimeoutMs": waitTimeoutMs,
+			})
```

Comment:
Minor: Instead of logging the wait timeout value inside the if block, can we log it after the block?

Reply:
Yep, either looks good to me. I can move this.

Reply:
Updated. (sry forgot to have this in a separate commit)